### PR TITLE
Simplify logging for file-system watching

### LINKED ISSUE #17396: Investigate `Detected overflow` in native logger
Related to https://gradle.slack.com/archives/C0WTFHWT1/p1618124196122000

Appears regularly, searching Slack I see:

* > Apr 19, 2021 10:46:43 AM net.rubygrapefruit.platform.internal.jni.NativeLogger log
INFO: Detected overflow for /Users/lorinc/IdeaProjects/dotcom
* https://gradle.slack.com/archives/C03A0MT6S/p1617742738187800?thread_ts=1617317203.149000&cid=C03A0MT6S
* https://gradle.slack.com/archives/C09LZKUCB/p1619014298003400?thread_ts=1618970464.001600&cid=C09LZKUCB
* https://gradle.slack.com/archives/C0WTFHWT1/p1615276741042000

---
cc: @gradle/execution

### PR DESCRIPTION
**Pull request by lptr**
This version uses SLF4J instead of JUL for logging.

We also dropped support for the `org.gradle.vfs.watch.debug` flag. This flag has never been documented. It wasn't something we used to get feedback from users for a long time, either. People wanting to observe FSW logs can use `--debug` instead.

Fixes #17396.

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

**Review by asodja — APPROVED** [review]
LGTM

File: subprojects/distributions-dependencies/build.gradle.kts [hunk 1]
```diff
@@ -73,7 +73,7 @@ dependencies {
         api(libs.eclipseSisuPlexus)     { version { strictly("0.3.5"); because("transitive dependency of Maven modules to process POM metadata") }}
         api(libs.errorProneAnnotations) { version { strictly("2.29.0") }}
         api(libs.fastutil)              { version { strictly("8.5.2") }}
-        api(libs.gradleFileEvents)      { version { strictly("0.2.3") }}
+        api(libs.gradleFileEvents)      { version { strictly("0.2.4-0") }}
```
[the remark this answers is not in the document]
Comment (lptr) [reply subprojects/distributions-dependencies/build.gradle.kts:76]: Yeah, I'm not happy about that, but given it's an internal library, I think it's okay for now. I added https://github.com/gradle/gradle-fileevents/issues/22 to clean it up later.

[the remark this answers is not in the document]
Comment (lptr) [reply subprojects/distributions-dependencies/build.gradle.kts]: Should be fixed now with `0.2.5`.

**Review by asodja — APPROVED** [review]
LGTM

**Pull request merged**